### ISSUE TITLE
Create requirements for new custom theme

### ISSUE THREAD

**Issue by kimisgold** [omeka/omeka-s-enduser#240]
We're spinning up our own MkDocs theme for our documentation for better control over the markup and our uses. You can find the first draft on the [`custom theme`](https://github.com/omeka/omeka-s-enduser/tree/custom-theme) branch. Please provide feedback for functional requirements you would like to see in the theme, especially with regards to accessibility. I'll start off the list with what I know still needs to be done.

- [x] Full tree navigation: currently the theme displays 1 level of nesting, so this pattern needs to be updated to support more levels.
- [x] Search results page needs customization, especially to remove repetitive search form.
- [x] Previous and Next pagination buttons
- [x] breadcrumb at top of page
- [x] page TOC integrated into left-hand page list (no righthand sidebar)
- [x] TOC stickied on page when scrolling
- [ ] hit highlighting when looking at search results
- [x] search results needs to be its own page with permalink
- [x] edit button takes you to corresponding github page
- [x] light/dark mode depending on system setting
- [x] admonitions (special visual styling for !!! note, etc.)
- [ ] tooltips, which we don't implement currently but the Material theme has
- [x] Also in terms of the navigation, an expand all/collapse all button might be welcome, depending on the direction the pattern goes in.
- [x] Screen reader results announcement, both for the dynamic loading as well as if this can be done on an initial search page load
- [x] Mobile menu is looking good in terms of focus order and dealing with obscured/visible content, but consistency of nav location should be ensured on mobile:
  - [x] On mobile only, a duplicate version of the nav (as in, in addition to that within the mobile nav modal) appears within top of main content, then promptly disappears.
  - [x] When viewport widens while modal is open, the container remains visible, but menu links disappear. If possible, the modal should disappear entirely, but I'm not set on the best location to then redirect focus since the initial nav toggle would not appear in a wider viewport.

**Comment by allanaaa** [comment]
- Previous and Next pagination buttons
- breadcrumb at top of page
- page TOC integrated into left-hand page list (no righthand sidebar)
- TOC stickied on page when scrolling
- hit highlighting when looking at search results
- search results needs to be its own page with permalink
- edit button takes you to corresponding github page
- light/dark mode depending on system setting
- admonitions (special visual styling for `!!! note`, etc.)
- tooltips, which we don't implement currently but the Material theme has

That's everything I turned on for the Material theme in our recent assessment there.

**Comment by aviannamiller** [comment]
@allanaaa has pretty much covered it, so I would just add the following for now:
- Also in terms of the navigation, an expand all/collapse all button might be welcome, depending on the direction the pattern goes in.
- Screen reader results announcement, both for the dynamic loading as well as if this can be done on an initial search page load
- Mobile menu is looking good in terms of focus order and dealing with obscured/visible content, but consistency of nav location should be ensured on mobile:
  - On mobile only, a duplicate version of the nav (as in, in addition to that within the mobile nav modal) appears within top of main content, then promptly disappears.
  - When viewport widens while modal is open, the container remains visible, but menu links disappear. If possible, the modal should disappear entirely, but I'm not set on the best location to then redirect focus since the initial nav toggle would not appear in a wider viewport.

**Comment by kimisgold** [comment]
There have been several updates to the theme. I've converted your requirements into a checklist, and it should reflect what I believe is currently addressed. Could you take a look and check for how things are feeling? Any new bugs you may be seeing? I think the main things left are search-related, as well as tooltips.

**Comment by allanaaa** [comment]
External links in the manual are just the usual markdown links with `{target=_blank}` appended to the end. This syntax appears to be unique to mkdocs. 

There should be a way for me to attach a class or style in Markdown that will render a link that indicates a new tab will be opened, as in the way we indicate external links in Omeka itself. Or, it may be that all absolute links (anything that starts with http) will automatically be identified as external and open in a new tab? That would be fine too. (I think the Material theme had some way to do this.)

If you can create a style, @kimisgold, that has a little "external" icon attached to it, I will figure out a way to apply it where needed.

**Comment by aviannamiller** [comment]
Also noting that along with a style to indicate external behavior, we would need to supply an accessible label that indicates the link opens in a new tab.

**Comment by kimisgold** [comment]
Yes, agreed with Avianna. I'm thinking for external links, we'd probably have to use a little HTML, something like

`[Omeka <span class="sr-only">external link</span>](https://omeka.org){target=_blank}`

I could still provide the external icon style, but that's what we'd need to make it screen reader readable.

**Comment by allanaaa** [comment]
My understanding is that Markdown can render some attributes, as mkdocs and others do. 

I hope to be able to append a class to a link with 

`[link text](linkurl){.external-link-class}`

And then your code would both render the visual icon / accessible label, and the new-tab behaviour.

**Comment by kimisgold** [comment]
> And then your code would both render the visual icon / accessible label, and the new-tab behaviour.

The problem is that there is no built-in means for screen readers to identify when a link will open a new window. For example:

<img width="610" height="199" alt="Screenshot of a link and its Voiceover readout" src="https://github.com/user-attachments/assets/cf241acd-00f9-4069-a1a0-f034c83bfa88" />

This is a Voiceover readout of a link to Omeka S's content on Vimeo. It uses `target="_blank"`, but screen readers do not use that attribute to indicate that the link opens a new window. 

With the screen reader only span example I mentioned, this is the kind of result we're trying to achieve.

<img width="1172" height="245" alt="Screenshot of a link, its Voiceover readout, and the web inspector showing updated markup." src="https://github.com/user-attachments/assets/cc5ae92f-e86f-46ae-b37a-5533d711cbd5" />

Here, the screen reader will pick up on the external link note without showing excess text to sighted users, who would have the benefit of a visual icon.

**Comment by aviannamiller** [comment]
> There have been several updates to the theme. I've converted your requirements into a checklist, and it should reflect what I believe is currently addressed. Could you take a look and check for how things are feeling? Any new bugs you may be seeing? I think the main things left are search-related, as well as tooltips.

@kimisgold, changes are working well! I only had a bit of time to spend with it so my feedback may not be all-encompassing, but I can take a closer look once you have new changes coming in, especially with the external link style or anything of that nature.

- [x] The `.icon-home` could either not have the underline or otherwise have an offset underline so it won't blend so much.
- [x] `<code>` is currently indistinguishable in style.
- [ ] /modules/collecting/#site-settings (see admonition block): Looks to be an issue with punctuation encoding, not sure if there are other places with this kind of problem.
- [x] Thoughts on making the pages listed within breadcrumbs nav clickable links? That would certainly boost accessibility by providing another mode of navigation, and it makes sense since they're tagged within a `<nav>` landmark anyway.
- [x] Search results levels skip from `<h1>` to `<h3>`.
- [x] Permanent links on `<h3>`s are 30x22.5px, should be at least 24px on each side. They continue to decrease in size when accompanying `<h4>`, `<h5>`, and so on.
- [x] `<h5>` font size is too small at 13.28px. Preferably, it should be at least 16px.
- [x] 13.3333px font size on Expand all / Collapse all buttons, would be at least 16px, preferably.
- [x] Can the full, non-anchor section headings (e.g. Sites, Resources, Admin) also be part of the expand/collapse interaction? I keep going to click them over the toggles themselves.
- [ ] We need to account for video (and potentially other) embeds; the iframes do not have accessible names. Can we automatically pull in media titles for these?
- [x] Mobile menu:
    - [x] Some bottom padding would be great. The last link is very close to the end of the container.
    - [x] Table of contents: Links nested under the active page link do not seem to be operable when the mobile nav modal is open. Clicking them does not do anything, and upon exiting the modal (which has to be done manually via the close button, unless navigating to an entirely different page within the manual), the page remains at the heading rather than the selected section.

**Comment by kimisgold** [comment]
I think most of the requests have been addressed. There are a few that I think lie outside of theme development, like accessible names for iframes and the punctuation rendering: I believe those need to be handled within the content itself.

If all the current implementations look good, only tooltips and search highlighting are left. I think I can figure out tooltips, but I think search highlighting will take more involved effort and discussion: figuring out snippet sizes, how many hits to highlight, how to organize multiple hits on a page, etc. I would make an issue for that feature to be implemented further down the line.

For my reference, what are the elements that need tooltips?

**Comment by allanaaa** [comment]
To add to the list / assess feasibility:

Content tabs, in the vein of https://squidfunk.github.io/mkdocs-material/reference/content-tabs/

**Comment by aviannamiller** [comment]
Thank you for all of the changes! I've verified that all the previously reported issues are fixed, while noting a few lingering areas from those for possible improvement:
- Can the breadcrumbs nav still include higher level links as a trail of the links? Right now, I'm just seeing a link for the home page and current page.
  - Speaking of the link to the home page, the `aria-label` on this is "Docs". Can we make this a bit more specific?
- Generally speaking, heading levels could maybe use more variation to help differentiate a bit more quickly between sections. I'm struggling to discern between heading levels for a page like Resource Templates, just to pluck out an example that includes all the way to h5.
> For my reference, what are the elements that need tooltips?

I'm not sure that there is a super specific current instance where tooltips are needed. The idea came about while reviewing some of the configurable features offered by [Material for MkDocs (Tooltips)](https://squidfunk.github.io/mkdocs-material/reference/tooltips/?h=tooltip). @allanaaa and I discussed some possible areas where these _could_ appear, such as to better define acronyms, perhaps link to the glossary in some way, for images, etc. If there is a specific area @allanaaa has in mind, we could certainly try it out, but this is more in the "could be nice to have" category and doesn't need to hold up anything for the redesign.
